pages/DSNP/Types/Update.md

```diff
@@ -0,0 +1,45 @@
+# Update Announcement
+
+An Update Announcement is a way to note intent to update previously
+announced content.
+
+## Fields
+
+| Field | Description | Data Type | Serialization | Parquet Type | Bloom Filter |
+| ----- | ----------- | --------- | ------------- | ------------ | ------------ |
+| announcementType | Announcement Type Enum (`6`) | enum | [decimal](../Serializations.md#decimal) | `INT32` | no |
+| createdAt | milliseconds since Unix epoch | 64 bit unsigned integer | [decimal](../Serializations.md#decimal) | `UINT_64` | no
+| fromId | id of the user creating the announcement | 64 bit unsigned integer | [decimal](../Serializations.md#decimal) | `UINT_64` | YES
+| contentHash | keccak-256 hash of updated content | 32 bytes | [hexadecimal](../Serializations.md#hexadecimal) | `BYTE_ARRAY` | YES
+| url | updated content URL | UTF-8 | [UTF-8](https://datatracker.ietf.org/doc/html/rfc3629) | `UTF8` | no
+| oldUrl | URL of old content | UTF-8 | [UTF-8](https://datatracker.ietf.org/doc/html/rfc3629) | `UTF8` | no
```

Comment:
- Shouldn't this be the old content hash not the old URL?
- I would also expect it would be in the bloom filter as people would be searching for updates to a particular hash from a given user.

Reply:
My understanding is that two users could, in theory, post the same content and so a contentHash by itself might not be enough to point to the target announcement. But the URI, again if I'm correct, would contain not only the content hash but a user's identifier as well, allowing us to precisely make an update.

But I realized in the process of writing this reply that if we assume that Updates can only be made to posts that have the same identifier as the `fromId`, then duplicate content hashes aren't really a concern. Can anyone think of a scenario where a user other than the original author would need to edit content?

Reply:
I cannot think of a case where any other `fromId` should be allowed to edit a post. Even if others might be moderating, that would not be an edit.

Reply:
Makes sense. One thing that jumps out at me about an implementation that relies on the`fromId` is a potential N+1 (we'll have to fetch the user's announcements and then lookup the contenthash). The nice thing about using the URI is it will be a single lookup. Does that sounds correct to you?